src/Http/Routing/src/Constraints/MinRouteConstraint.cs

```diff
@@ -28,11 +36,17 @@ public MinRouteConstraint(long min)
 
     /// <inheritdoc />
     public bool Match(
+#if !COMPONENTS
         HttpContext? httpContext,
         IRouter? route,
+#endif
         string routeKey,
+#if !COMPONENTS
         RouteValueDictionary values,
         RouteDirection routeDirection)
+#else
+        RouteValueDictionary values)
+#endif
```

Comment:
Who would you say is most responsible for maintaining this? I don't recall ever touching this code myself so am not clear on what level of maintainability is required here and what is the likelihood of nontrivial work happening in this area ever again.

This level of intricacy of `#if` over dozens of method signatures is not something I'd normally consider the way we do things, and is definitely suggestive that we've picked the wrong abstraction and should find a way to rework it. Clearly it would be better if the shared parts could be factored into a self-contained common package with suitable abstractions instead of the many `#if`s.

However if you're taking the position that (1) this doesn't impact any other team, this is only ours to maintain and (2) there's no reason to think this is going to be developed any further in the forseeable future, then perhaps it's rational to accept the technical debt here (on the basis that we think nothing will ever force us to make repayments).

Reply:
@SteveSandersonMS that would be me. 

It's not likely for significant work to happen in this area (maybe we add an extra constraint, but that's it) and we can do the refactor in 9.0 if we need to.

I'm already thinking of reworking the constraints into something more modern and suitable for both, Blazor and ASP.NET Core, as they are legacy from the old routing times (if you see, they take an IRouter instance, which predates endpoint routing, but its there for legacy/compatibility). That said, it would be work for 9.0